crates/proto/src/serialize/txt/parse_rdata.rs

```diff
@@ -88,6 +88,9 @@ impl RDataParser for RData {
             RecordType::TLSA => Self::TLSA(tlsa::parse(tokens)?),
             RecordType::TXT => Self::TXT(txt::parse(tokens)?),
             RecordType::SIG => return Err(ParseError::from("parsing SIG doesn't make sense")),
+            #[cfg(feature = "dnssec")]
+            RecordType::DNSKEY => Self::DNSSEC(DNSSECRData::DNSKEY(dnskey::parse(tokens)?)),
```

Comment:
IIUC, hickory-dns is not meant to work with pre-signed zone files like `nsd` does so when this `Parser` API is used to load a zone file into a "file" authority we should continue to reject DNSKEY records.

IOW, I should probably add some flag here that determines whether DNSKEY records should be accepted or not. this `Parser` is also used to parse a "trust anchor" file which is a newline separated list of DNSKEY records; in that case we want the `Parser` to (only) parse DNSKEY records. when we use `Parser` in service of a file authority we want to reject DNSKEY records like the original code was doing.

Reply:
In our usage, is there any case where we need to parse both `DNSKEY` records and other kinds of records? If not, we should just split the parser?

Reply:
Well, `unbound` supports trust anchor files that contain both `DNSKEY` and `DS` records. Meaning that `unbound` can complete DNSSEC validation with just a DS record in the trust anchor file.

In their current state, `DnssecDnsHandle` and `TrustAnchor` only support DNSSEC validation using public key material (`DNSKEY` information) so a trust anchor file with just DS records would not work and should be rejected.

So, yeah. It's probably best to have separate parsers: the general `Parser` for regular zone files and a `TrustAnchorParser` that only accepts newline separated DNSKEY records (and maybe also DS records in the future).

Reply:
I have submitted this whole separate parser idea in #2257